### ISSUE TITLE
Gatsby images cannot be loaded wrapped by <Zoom><Zoom/>

### ISSUE THREAD

**Issue by Sherry2016** [rpearce/image-zoom#255]
This issue is a:
- Question / support request

Hello! Thanks for develop and maintaining this package :)
I have an issue trying to use it with gatsby images. Not sure if the <StaticQuery /> would make any difference on the zoom, but when I try to do something like
```tsx
              <StaticQuery
                query={graphql`
                  query imageQuery {
                    source: allFile(filter: { absolutePath: { regex: "/images/" }}) {
                      edges {
                        node {
                          childImageSharp {
                            fluid(maxWidth: 256) {
                              ...GatsbyImageSharpFluid
                            }
                          }
                        }
                      }
                    }
                  }
                `}
                render={data => (
                  <div>
                    {data.source.edges.map(({ node }, i) => (
                      <div key={i} sx={{marginTop: '1rem'}}>
                        <Zoom>
                          <Img fluid={node.childImageSharp.fluid} />
                        </Zoom>  
                      </div>
                    ))}
                  </div>
                )}
              />
```
No images would render. 
But images would show up without the <Zoom />. I've also tried <ControlledZoom /> which also doesn't work. 
Also not sure if it's other formatting that messed up the zoom. I'm also using animated.div from react-spring.

Any help is appreciated. Thanks in advance!

**Comment by rpearce** [comment]
Hi! Thanks for opening this issue.

> But images would show up with out the .

👆 What do you mean by this?

* * *

People have been using it with gatsby, and a reproduction of the issue would be helpful. However, perhaps this is a good opportunity for you to try out the next version, https://www.npmjs.com/package/react-medium-image-zoom/v/5.0.0-alpha.9, that is close to shipping?

There are gatsby instructions here for working with it: https://github.com/rpearce/image-zoom/issues/227#issuecomment-636888606.

* * *

Let me know if I can help further.

**Comment by Sherry2016** [comment]
Thanks for the reply. Sorry for the typo. I meant that the images would show up without the <Zoom></Zoom> wrapper.
I would try the new version and let you know.

**Comment by Sherry2016** [comment]
Hi @rpearce  I just tried the new version. There is some progress :) The images can load now, however only the last one image in the list is zoomable. There is no reaction when I hover over other images. I'm doing it like this now
```tsx
              <StaticQuery
                query={graphql`
                  query imageQuery {
                    source: allFile(filter: { absolutePath: { regex: "/images/" }}, sort: { fields: [name], order: ASC}) {
                      edges {
                        node {
                          childImageSharp {
                            fluid(maxWidth: 2560) {
                              ...GatsbyImageSharpFluid
                            }
                          },
                        }
                      }
                    }
                  }
                `}
                render={data => (
                  <div>
                    {data.source.edges.map(({ node }, i) => (
                      <div ref={ref} key={i} sx={{marginTop: '1rem'}}>
                        <Img fluid={node.childImageSharp.fluid} />
                      </div>
                    ))}
                  </div>
                )}
              />
```

**Comment by rpearce** [comment]
I think you need a new ref in that `render` function for every item:
```js
                render={data => {
                  const { ref } = useImageZoom()
                  return (
                    <div>
                      {data.source.edges.map(({ node }, i) => (
                        <div ref={ref} key={i} sx={{marginTop: '1rem'}}>
                          <Img fluid={node.childImageSharp.fluid} />
                        </div>
                      ))}
                    </div>
                  )}
```

**Comment by Sherry2016** [comment]
Yes!! Thank you very much for the help!!! Now it works like a charm! :D 
(Although the hook should be inside the map loop to work :)

**Comment by rpearce** [comment]
That's great. Keep an eye out for when the official release for `v5.0` comes soon.

**Comment by Sherry2016** [comment]
👍

**Comment by tarviroos** [comment]
Sorry for bringing up an old issue. :) However, it seems like a good place to ask this question.

With the newly released `v5.0` (congratz btw 🎉) there is no useImageZoom, right? I'm just looking into making it work with GatsbyImage. Although I get the image to display nicely by wrapping it with <Zoom /> component, the zooming functionality is not there. Is there something I'm missing?

**Comment by rpearce** [comment]
@tarviroos Thats right; the hook was an experiment that never came to fruition. The README should accurately detail what needs doing: import the library, import the CSS (or copy and paste it and use/alter it yourself).

I tested this in a basic, regular Gatsby project just the other day. Could you be so kind as to share an example project where it's not working so that I can help you?

I'd also create a separate issue.
